sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -43,6 +43,21 @@ private[sql] class JSONOptions(
 
   import JSONOptions._
 
+  private val maxNestingDepth: Int = parameters
```

Comment:
don't make any changes yet - but @LuciferYang, are there are guidelines about to handle cases where the parameter values are not convertible to ints (.toInt fails) and where the values evaluate to negative values (which will cause an IllegalArgumentException when setting up StreamReadConstraints

Reply:
I think this is fine. It's a niche set of configs that very few would set. If they set a bad value, a raw error is even OK.